packages/frontend/src/shared/remark/remark-iex.ts

```diff
@@ -33,153 +31,209 @@ export const remarkIex = (options) => {
+            // Apply attributes to nested table node
+            // This directive node will not be rendered
+            node.children[0].data = {
+              hName: 'table',
+              hProperties: node.attributes
+            };
+          }
         }
+
         break;
       }
-      case 'katex':
-        node.type = 'math';
-        node.value = remarkNodetoText(node);
-        break;
-      case 'vega': {
-        node.type = 'vegaChart';
 
-        // Extract body of node into text
-        node.config = remarkNodetoText(node);
+      case 'katex': {
+        node.data = {
+          hName: 'katex',
+          hProperties: {
+            math: remarkNodetoText(node),
+            block: 'true'
+          }
+        };
 
-        // Manually calculate the key to avoid costly redraws
-        // when this node isn't modified
-        node.key = hashCode(node.config);
+        break;
+      }
 
+      case 'vega': {
         // Convert height/width if available
         const { height: h, width: w, ...attributes } = node.attributes;
-        node.attributes = {
-          w,
-          h,
-          ...attributes
+
+        node.data = {
+          hName: 'vegachart',
+          hProperties: {
+            // Extract body of node into text
+            config: remarkNodetoText(node),
+            w,
+            h,
+            ...attributes
+          }
         };
 
         delete node.children;
 
         break;
       }
+
       case 'xkcd': {
-        //console.log(node);
-        node.type = 'xkcdChart';
+        let xkcdType = 'line';
 
         // First child represents the bracket contents
         // and should have `directiveLabel` set
         // e.g. :::xkcd[type]
         // Parse out the contents and remove that child
         if (node.children && node.children.length > 0) {
           if (node.children[0].data?.directiveLabel === true) {
-            node.xkcdType = remarkNodetoText(node.children[0]);
+            xkcdType = remarkNodetoText(node.children[0]);
             node.children.shift();
           }
         }
 
-        // Extract body of node into text
-        node.config = remarkNodetoText(node);
-
-        // Manually calculate the key to avoid costly redraws
-        // when this node isn't modified
-        node.key = hashCode(node.config);
-
         // Convert height/width if available
         const { height: h, width: w, ...attributes } = node.attributes;
-        node.attributes = {
-          w,
-          h,
-          ...attributes
+
+        node.data = {
+          hName: 'xkcdchart',
+          hProperties: {
+            // Extract body of node into text
+            config: remarkNodetoText(node),
+            xkcdType,
+            w,
+            h,
+            ...attributes
+          }
         };
 
         delete node.children;
 
         break;
       }
+
+      default:
+        node.data = {
+          hName: 'containerdirective',
+          hProperties: node
+        };
```

Comment:
Unlike before, we don't get a `containerDirective` node calling our renderer, so we need to make one manually here.